scipy/sparse/csgraph/_shortest_path.pyx

```diff
@@ -630,6 +630,8 @@ cdef _dijkstra_setup_heap_multi(FibonacciHeap *heap,
     heap.min_node = NULL
     for i in range(Nind):
         j_source = source_indices[i]
+        if nodes[j_source].state == SCANNED:
```

Comment:
This might be clearer if you move the `current_node = &nodes[j_source]` line up above this one, then refer to `current_node.state`.